lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2386,10 +2386,10 @@ buildTransactionPure
                 (Left $ unsafeShelleyOnlyGetRewardXPub @s (getState wallet))
                 txCtx
                 (Left preSelection)
+    let utxoAll = Write.fromWalletUTxO utxo
+    let utxoSpendable = Map.keysSet (unUTxO utxoAll)
```

Comment:
In the case where there is no partial transaction available, we allow `balanceTx` to spend from the entire UTxO set.